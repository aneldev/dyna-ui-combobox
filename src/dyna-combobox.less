.dyna-combobox {
	.dyna-ui-label{
		cursor: default;
	}

	.dyna-ui-field-wrapper-container {
		padding: 0 10px 0 0 !important;
	}

	&.dyna-ui-field-wrapper.dyna-ui-field-wrapper--size- {
		&XSMALL {
			select {
				padding: 9px;
			}
		}
		&SMALL {
			select {
				padding: 9px;
			}
		}
		&MEDIUM {
			select {
				padding: 16px;
			}
		}
		&LARGE {
			select {
				padding: 20px;
				font-size: 20px;

			}
		}
	}

	&.is-mac-chrome-safari.dyna-ui-field-wrapper.dyna-ui-field-wrapper--size- {
		&XSMALL {
			select {
				padding: 0;
				margin: 9px 0;
			}
		}
		&SMALL {
			select {
				padding: 0;
				margin: 9px 0;
			}
		}
		&MEDIUM {
			select {
				padding: 0;
				margin: 16px 0;
			}
		}
		&LARGE {
			select {
				padding: 0;
				margin: 20px 0;
			}
		}
	}

	select {
		border: 0;
		cursor: pointer;
		background-color: transparent;

		&:focus {
			outline: none;
		}
	}
}
