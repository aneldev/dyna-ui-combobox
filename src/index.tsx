import {
  EStyle, EColor, ESize,
} from "dyna-ui-field-wrapper";

import {
  DynaComboBox, IDynaComboBoxProps,
} from "./DynaComboBox";

export {
  DynaComboBox, IDynaComboBoxProps,
  EStyle, EColor, ESize,
}
