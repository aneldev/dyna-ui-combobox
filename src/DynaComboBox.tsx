import * as React from "react";

import {EMode, EColor, EStyle, ESize, DynaFieldWrapper} from "dyna-ui-field-wrapper";

import "./dyna-combobox.less";

export interface IDynaComboBoxProps {
  name: string;
  label?: JSX.Element | string;
  mode?: EMode,
  style?: EStyle;
  color?: EColor;
  size?: ESize;
  options?: IDynaComboBoxOption[];
  value?: string;
  onChange?: (name: string, value: string) => void;
}

export interface IDynaComboBoxOption {
  label: string;
  value: string;
}

export class DynaComboBox extends React.Component<IDynaComboBoxProps> {
  static defaultProps: IDynaComboBoxProps = {
    name: null,
    label: null,
    mode: EMode.EDIT,
    style: EStyle.INLINE_ROUNDED,
    color: EColor.WHITE_BLACK,
    size: ESize.MEDIUM,
    options: [],
    value: null,
    onChange: (name: string, value: string) => undefined,
  };

  private handleChange(event: any): void {
    const {name, onChange} = this.props;
    onChange(name, event.target.value);
  }

  public render(): JSX.Element {
    const {
      label,
      mode,
      style, color, size,
      options,
      value,
    } = this.props;

    return (
      <DynaFieldWrapper
        className="dyna-combobox"
        mode={mode}
        style={style}
        color={color}
        size={size}
        label={label}
        inputElementSelector="select"
      >
        <select
          value={value || ''}
          onChange={this.handleChange.bind(this)}
        >
          {options.map((option: IDynaComboBoxOption, index: number) => {
            return (
              <option
                key={index}
                value={option.value}
              >{option.label}</option>
            );
          })}
        </select>
      </DynaFieldWrapper>
    );
  }
}
