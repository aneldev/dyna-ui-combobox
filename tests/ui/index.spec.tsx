import "jest";

import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import * as React from 'react';
import * as enzyme from 'enzyme';

import {DynaComboBox} from '../../src';

describe('Home', () => {
  let wrapper;

  it('has expected content with deep render', () => {
    wrapper = enzyme.shallow(
      (
        <DynaComboBox
        name="currency"
        label="Currency"
        value="eur"
        options={[
          {value: 'eur', label: 'Euro'},
          {value: 'usd', label: 'US Dollar'},
          {value: 'aud', label: 'Australian Dollar'},
          {value: 'gbp', label: 'British Pound'},
        ]}
        />
      ),
      {}
    );

    expect(wrapper).toMatchSnapshot();
  });
});
