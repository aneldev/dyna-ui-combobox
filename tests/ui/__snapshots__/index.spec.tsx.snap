// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has expected content with deep render 1`] = `
<DynaFieldWrapper
  className="dyna-combobox"
  color="WHITE_BLACK"
  footer={null}
  inputElementSelector="select"
  isLoading={null}
  label="Currency"
  mode="EDIT"
  onFocus={[Function]}
  required={null}
  size="MEDIUM"
  style="INLINE_ROUNDED"
  validationMessage={null}
>
  <select
    onChange={[Function]}
    value="eur"
>
    <option
      key="0"
      value="eur"
    >
      Euro
    </option>
    <option
      key="1"
      value="usd"
    >
      US Dollar
    </option>
    <option
      key="2"
      value="aud"
    >
      Australian Dollar
    </option>
    <option
      key="3"
      value="gbp"
    >
      British Pound
    </option>
  </select>
</DynaFieldWrapper>
`;
